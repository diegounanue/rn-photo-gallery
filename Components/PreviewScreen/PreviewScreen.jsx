import React, { useState, useEffect, useContext } from 'react';
import { StyleSheet ,Text, View, Button, Image, Dimensions} from 'react-native';
import { Camera } from 'expo-camera';
import { PhotoGalleryContext } from "../../Contexts/PhotoGalleryContext";

export default function PreviewScreen({ route, navigation }) {
  const [hasCameraPermission] = useState(null);
  const { addPhotoToGallery } = useContext(PhotoGalleryContext);


  const { photoUri } = route.params;

  const handleAccept = () => {
    addPhotoToGallery(photoUri)
    navigation.navigate('Home'); 
  };

  const handleReject = () => {
    navigation.goBack();
  };

  return (
    <View style={styles.container}>
    <Image source={{ uri: photoUri }} style={styles.image} />
    <View style={styles.buttonContainer}>
      <Button title="Accept" onPress={handleAccept} />
      <Button title="Reject" onPress={handleReject} />
    </View>
  </View>
  );
}

const { height } = Dimensions.get('window');

const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
    },
    image: {
      width: '90%',
      height: height * 0.8 , // El 90% de la altura de la pantalla
      resizeMode: 'cover',
    },
    buttonContainer: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      width: '80%',
      marginTop: 20,
    },
  });