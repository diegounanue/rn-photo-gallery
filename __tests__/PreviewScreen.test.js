import React from 'react';
import renderer from 'react-test-renderer';
import PreviewScreen from '../Components/PreviewScreen/PreviewScreen';

describe('PreviewScreen Component', () => {
  it('renders correctly', () => {
    const route = {
      params: {
        photoUri: 'https://example.com/photo.jpg', // Reemplaza con una URL válida
      },
    };
    const navigation = {
      navigate: jest.fn(),
      goBack: jest.fn(),
    };

    const tree = renderer
      .create(<PreviewScreen route={route} navigation={navigation} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls handleAccept correctly', () => {
    const route = {
      params: {
        photoUri: 'https://example.com/photo.jpg', // Reemplaza con una URL válida
      },
    };
    const navigation = {
      navigate: jest.fn(),
      goBack: jest.fn(),
    };

    const component = renderer.create(
      <PreviewScreen route={route} navigation={navigation} />
    );

    const instance = component.root;
    const handleAcceptButton = instance.findByProps({ title: 'Accept' });

    handleAcceptButton.props.onPress(); // Simula hacer clic en el botón "Accept"

    expect(navigation.navigate).toHaveBeenCalledWith('Home');
  });

  it('calls handleReject correctly', () => {
    const route = {
      params: {
        photoUri: 'https://example.com/photo.jpg', // Reemplaza con una URL válida
      },
    };
    const navigation = {
      navigate: jest.fn(),
      goBack: jest.fn(),
    };

    const component = renderer.create(
      <PreviewScreen route={route} navigation={navigation} />
    );

    const instance = component.root;
    const handleRejectButton = instance.findByProps({ title: 'Reject' });

    handleRejectButton.props.onPress(); // Simula hacer clic en el botón "Reject"

    expect(navigation.goBack).toHaveBeenCalled();
  });
});
