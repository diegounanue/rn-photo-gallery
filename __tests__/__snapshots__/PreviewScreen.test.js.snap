// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewScreen Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      {
        "uri": "https://example.com/photo.jpg",
      }
    }
    style={
      {
        "height": 1067.2,
        "resizeMode": "cover",
        "width": "90%",
      }
    }
    testID="preview-image"
  />
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 20,
        "width": "80%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Accept
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Reject
        </Text>
      </View>
    </View>
  </View>
</View>
`;
